tests/test_pool.py

```diff
@@ -62,6 +62,7 @@ def get_generator(iterable):
             yield from iterable
 
         # Test results for different parameter settings
+        print()
```

Comment:
Yeah, by default I added some printing in the tests such that you can check progress. For some tests, especially those surrounding exceptions, it's been really helpful to know in which subtest it is when it e.g. deadlocked on github. 

Normal `pytest -v .` will supress those print statements.

Do you see an issue with these statements?

Reply:
Well, it seems a bit odd. Maybe it's also an indication the tests are too complicated. Splitting things into multiple tests would allow the normal way for pytest to show progress to do some more of the work. But that won't work in cases where you want to loop over a lot of options/settings.

Reply:
Using pytest.mark.parametrize, you can probably also eliminate some cases that otherwise require subtests to loop over inputs.
https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
Thnx for the link. I don't have much experience with the different pytest features so will check it out!

Reply:
I guess I can't use parameterize from pytest together with unittest.TestCase? That's unfortunate. This means I will have to rewrite all tests to not use classes. Or is there a way that it does work together?

If I go that route, do you have suggestions on how to cope with:
- A TestCase.setup() equivalent for setting up data for each test function?
- Grouping logically related tests. Or should I just prefix the test function and use comments in the file to indicate a new section of tests? I.e., what is common practice?

Reply:
Hmm, that's annoying. I'd think it shouldn't have been that hard for them to write parametrize in a way that works for TestCase methods. 
Maybe this is also something better left as improvement in another issue?

Reply:
NB it looks like you can use normal classes with pytest

```
import pytest

class TestCase:
   @pytest.mark.parametrize("test_input,expected", [("3+5", 8), ("2+4", 6), ("6*9", 42)])
   def test_eval(self, test_input, expected):
       assert eval(test_input) == expected
```
There's also ways to setup/teardown classes and tests: https://docs.pytest.org/en/6.2.x/xunit_setup.html

It would still require changing all the asserts as well, though.

Reply:
Ah nice. So structurally not a lot needs to change then. But I agree, let's do that in another PR